app/models/concerns/foreman_ansible/host_managed_extensions.rb

```diff
@@ -4,51 +4,69 @@
+        scoped_search :relation => :ansible_roles, :on => :name,
+                      :complete_value => true, :rename => :role,
+                      :only_explicit => true
 
-      def inherited_ansible_roles
-        return [] unless hostgroup
-        hostgroup.inherited_and_own_ansible_roles
+        before_provision :play_ansible_roles
+        audit_associations :ansible_roles
       end
 
-      # This one should be fixed, disabled for the moment as we're
-      # in a rush to get the release out
-      # rubocop:disable Metrics/AbcSize
-      def play_ansible_roles
-        return true unless ansible_roles.present? ||
-                           inherited_ansible_roles.present?
-        composer = job_composer(:ansible_run_host, self)
-        composer.triggering.mode = :future
-        composer.triggering.start_at = (
-          Time.zone.now +
-          Setting::Ansible[:ansible_post_provision_timeout].to_i.seconds
-        )
-        composer.trigger!
-        logger.info("Task for Ansible roles on #{self} before_provision: "\
-                    "#{job_invocation_path(composer.job_invocation)}")
-      rescue Foreman::Exception => e
-        logger.info("Error running Ansible roles on #{self} before_provision: "\
-                    "#{e.message}")
-      end
-      # rubocop:enable Metrics/AbcSize
+      base.singleton_class.prepend ClassMethods
+    end
+
+    def inherited_ansible_roles
+      return [] unless hostgroup
+      hostgroup.inherited_and_own_ansible_roles
+    end
+
+    # This one should be fixed, disabled for the moment as we're
+    # in a rush to get the release out
+    # rubocop:disable Metrics/AbcSize
+    def play_ansible_roles
+      return true unless ansible_roles.present? ||
+                         inherited_ansible_roles.present?
+      composer = job_composer(:ansible_run_host, self)
+      composer.triggering.mode = :future
+      composer.triggering.start_at = (
+        Time.zone.now +
+        Setting::Ansible[:ansible_post_provision_timeout].to_i.seconds
+      )
+      composer.trigger!
+      logger.info("Task for Ansible roles on #{self} before_provision: "\
+                  "#{job_invocation_path(composer.job_invocation)}")
+    rescue Foreman::Exception => e
+      logger.info("Error running Ansible roles on #{self} before_provision: "\
+                  "#{e.message}")
+    end
+    # rubocop:enable Metrics/AbcSize
 
-      def all_ansible_roles
-        (ansible_roles + inherited_ansible_roles).uniq
+    def all_ansible_roles
+      (ansible_roles + inherited_ansible_roles).uniq
+    end
+
+    def host_params_hash
+      values = AnsibleVariable.where(:ansible_role_id => all_ansible_roles.pluck(:id)).
+               values_hash(self).raw.values
+
+      transformed = values.each_with_object({}) do |item, memo|
+        item.map do |name, hash|
+          memo[name] = { :value => hash[:value], :safe_value => hash[:value], :source => 'global' }
```

Comment:
should the source link the specific ansible variable? I suppose that would mean introducing new Parameter type from this plugin though, so global may be ok here for now

safe_value though is more interesting, this means we'll always display the value right? even if the ansible variable is marked as hidden, we should probably use https://github.com/theforeman/foreman/blob/develop/app/models/concerns/hidden_value.rb here